overlays/nodejs/serverless/function.ts

```diff
@@ -83,6 +83,10 @@ export interface FunctionOptions {
         securityGroupIds: pulumi.Input<string[]>,
         subnetIds: pulumi.Input<string[]>,
     };
+    /**
+     * The Lambda environment's configuration settings.
+     */
+    environment?: {[key: string]: string}
```

Comment:
We haven't been consistent about this in other types here, but you could make this flexible to use with outputs from other Pulumi resources by using the same type (with `pulumi.Input` at each layer) as on the underlying `aws.lambda.FunctionArgs`:

```
environment?: pulumi.Input<{ variables?: pulumi.Input<{[key: string]: pulumi.Input<string>}> }>;
```